File: admin_backend/loginregister/src/main/java/com/example/loginregister/model/Payment.java
// src/main/java/com/example/loginregister/model/Payment.java
package com.example.loginregister.model;

import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;

@Entity
public class Payment {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    private String name;
    private String email;
    private String address;
    private String city;
    private String state;
    private String postalCode;
    private String cardNumber;
    private String expiryDate;
    private String cvv;
    private Double amount;

    // Default constructor
    public Payment() {}

    // Parameterized constructor
    public Payment(String name, String email, String address, String city, String state, String postalCode, String cardNumber, String expiryDate, String cvv, Double amount) {
        this.name = name;
        this.email = email;
        this.address = address;
        this.city = city;
        this.state = state;
        this.postalCode = postalCode;
        this.cardNumber = cardNumber;
        this.expiryDate = expiryDate;
        this.cvv = cvv;
        this.amount = amount;
    }

    // Getters and Setters
    public Long getId() { return id; }
    public void setId(Long id) { this.id = id; }

    public String getName() { return name; }
    public void setName(String name) { this.name = name; }

    public String getEmail() { return email; }
    public void setEmail(String email) { this.email = email; }

    public String getAddress() { return address; }
    public void setAddress(String address) { this.address = address; }

    public String getCity() { return city; }
    public void setCity(String city) { this.city = city; }

    public String getState() { return state; }
    public void setState(String state) { this.state = state; }

    public String getPostalCode() { return postalCode; }
    public void setPostalCode(String postalCode) { this.postalCode = postalCode; }

    public String getCardNumber() { return cardNumber; }
    public void setCardNumber(String cardNumber) { this.cardNumber = cardNumber; }

    public String getExpiryDate() { return expiryDate; }
    public void setExpiryDate(String expiryDate) { this.expiryDate = expiryDate; }

    public String getCvv() { return cvv; }
    public void setCvv(String cvv) { this.cvv = cvv; }

    public Double getAmount() { return amount; }
    public void setAmount(Double amount) { this.amount = amount; }
}

File: user_frontend/src/model/Payment.js
// src/models/Payment.js
class Payment {
  constructor(
    name,
    email,
    address,
    city,
    state,
    postalCode,
    cardNumber,
    expiryDate,
    cvv,
    amount
  ) {
    this.name = name;
    this.email = email;
    this.address = address;
    this.city = city;
    this.state = state;
    this.postalCode = postalCode;
    this.cardNumber = cardNumber;
    this.expiryDate = expiryDate;
    this.cvv = cvv;
    this.amount = amount;
  }

  validateCardNumber() {
    return /^\d{16}$/.test(this.cardNumber);
  }

  validateCVV() {
    return /^\d{3,4}$/.test(this.cvv);
  }

  validateExpiryDate() {
    const [month, year] = this.expiryDate.split("/");
    if (!month || !year) return false;
    const expiry = new Date(`20${year}-${month}-01`);
    return expiry > new Date();
  }

  validateEmail() {
    // Implementasi sederhana untuk validasi email
    return /^[^\s@]+@[^\s@]+\.[^\s@]+$/.test(this.email);
  }

  validate() {
    return (
      this.validateCardNumber() &&
      this.validateCVV() &&
      this.validateExpiryDate() &&
      this.validateEmail()
    );
  }
}

export default Payment;
